### PR TITLE
Merge testing into master (migration to PostgreSQL, change arm64 source, Matermost v7.5.2)

### PR DESCRIPTION
**Pull request by ericgaspar**
- Migration to PostgrSQL #377
- Change ARM source #409

### CODE REVIEW

**Comment by ericgaspar** [comment]
!testme

**Comment by kay0u** [comment]
From #377 

> I think it’s okay to merge this into `testing`.
> 
> But IMO, before releasing it, we should still handle the case of the Focalboard plug-in on 7.3 installation, which in the current state is going to break the upgrade on many 7.3 systems.

@kemenaran how can i generate this error?

**Comment by kemenaran** [comment]
@kay0u I think the steps would be:

1. Install Mattermost 7.2 on Yunohost, using the current master
2. Ensure the Boards plug-in is installed and activated (otherwise install it from Mattermust’s Marketplace)
3. Upgrade to Mattermost 7.3, using this commit : https://github.com/YunoHost-Apps/mattermost_ynh/commit/1677cb9fa9d95ab788a6acd5b310b09c7d2ab7df (which was available for a few days, then reverted)
4. The Boards plug-ins should be disabled, because its migrations failed on MariaDB. The schema of the focalboards table will stay on the 7.2 version (instead of 7.3).
5. Upgrade to Mattermost 7.5, using the current testing branch (This will attempt to migrate from MariaDB to Postgres)

At this stage, the pgloader migration should fail entirely, when it attempts to load the focalboard_* tables from a 7.2 schema (on MariaDB) to a 7.3 schema (on Postgres).

This is an issue because the Boards plug-in went pre-packaged with Mattermost at some point, and so was installed by default.

**Comment by kemenaran** [comment]
@kay0u otherwise, some simpler steps to simulate the same issue:

1. Install Mattermost 7.2 or 7.3
2. Stop the Mattermost server
3. Connect to the MySQL database manually, and create a column `focalboard_blocks.workspace_id` with a single value
4. Try to migrate to Mattermost 7.5

This should fail in the same way (because the column `focalboard_blocks.workspace_id` doesn't exist in the target schema).

File: scripts/upgrade [hunk 1]
```diff
@@ -56,6 +56,10 @@ ynh_script_progression --message="Backing up the app before upgrading (may take
 # Backup the current version of the app
 ynh_backup_before_upgrade
 ynh_clean_setup () {
+  # Remove the PostgreSQL db in case of error during the migration.
+  if [ -n "$remove_psql_in_case_of_error" ]; then
+    ynh_psql_remove_db --db_user=$db_user --db_name=$db_name
+  fi
```
Comment (kemenaran) [inline scripts/upgrade:62]: Yay, the Postgres DB not being deleted caused me a lot of hard-to-debug issues, thanks for this!

**Comment by kay0u** [comment]
¯\_(ツ)_/¯ This makes the migration from 7.3.0 to 7.5.2 pass on my side

**Comment by kemenaran** [comment]
@kay0u yay! I tried to play with pgloader’s `EXCLUDING TABLES MATCHING s/focalboard_/`, but couldn’t get it to work. I guess the brute-force solution is good too. Thanks!

**Comment by kemenaran** [comment]
I wrote a [request for testing](https://forum.yunohost.org/t/need-testers-for-the-latest-mattermost-upgrade/23992) on the Yunohost forums; I guess we can merge this after a few successful migration reports.

**Comment by kay0u** [comment]
> @kay0u yay! I tried to play with pgloader’s `EXCLUDING TABLES MATCHING s/focalboard_/`, but couldn’t get it to work. I guess the brute-force solution is good too. Thanks!

I managed to make it work:
```
EXCLUDING TABLE NAMES MATCHING ~/^focalboard_/
```

**Comment by ericgaspar** [comment]
shall we merge?

**Review by kemenaran — APPROVED** [review]
There was like 3 testers that reported an upgrade on a real instance completing successfully. That doesn’t seem a lot to me, but I don’t really see how to get more testing at this stage.

Let’s go! At least the upgrade is safe, so if it fails we can still push a fix later.

**Pull request merged**